test/framework/src/main/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java

```diff
@@ -582,7 +538,7 @@ protected SnapshotInfo getSnapshot(String repository, String snapshot) {
 
     protected void awaitMasterFinishRepoOperations() throws Exception {
         logger.info("--> waiting for master to finish all repo operations on its SNAPSHOT pool");
-        final ThreadPool masterThreadPool = internalCluster().getMasterNodeInstance(ThreadPool.class);
+        final ThreadPool masterThreadPool = internalCluster().getCurrentMasterNodeInstance(ThreadPool.class);
```

Comment:
This actually fixes a bug, we just didn't see the bug in practice yet because we only used this method in tests with a single master node ...

Reply:
Yeah, I noticed similar issue in DiskThresholdDeciderIT. getMasterNodeInstance() is trappy and should be renamed.